sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -111,6 +111,9 @@ int coap_parse(coap_pkt_t *pkt, uint8_t *buf, size_t len)
                 DEBUG("optpos option_nr=%u %u\n", (unsigned)option_nr, (unsigned)optpos->offset);
                 optpos++;
                 option_count++;
+                if (option_count >= NANOCOAP_NOPTS_MAX) {
```

Comment:
It should be OK to include NANOCOAP_NOPTS_MAX options. This test should be moved, above addition to this array, probably line 109. The test then should be:

`if (option_count == NANOCOAP_NOPTS_MAX) {`

Reply:
Yes. The new test seems to be working, it exposed that using "==" here caused to write one behind the options array. So I moved the check to the beginning of the if case.